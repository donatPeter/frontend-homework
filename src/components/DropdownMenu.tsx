import React from "react";
import MenuLink from "./MenuLink";
import Companies from "./Companies";
import styled from "styled-components";

const DropdownMenuContainer = styled.div`
  width: 304px;
  background: #ffffff;
  border: 1px solid #dcdce0;
  box-sizing: border-box;
  box-shadow: 0 2px 8px rgba(51, 51, 51, 0.24);
  border-radius: 8px;

  position: absolute;
  top: 65px;
`;

const DropdownMenuList = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
`;

const LinksContainer = styled.div`
  padding: 8px 0;
  width: 100%;
  border-bottom: 1px solid #dcdce0;
`;

const DropdownMenu = () => (
  <DropdownMenuContainer>
    <DropdownMenuList>
      <Companies />

      <LinksContainer>
        <MenuLink
          text="Get the mobile app"
          icon="phone_iphone"
          href="/get-app"
        />
        <MenuLink text="Community" icon="people" href="/community" />
        <MenuLink text="Knowledge base" icon="book" href="/help" />
      </LinksContainer>

      <LinksContainer>
        <MenuLink text="Settings" icon="settings" href="/settings" />
        <MenuLink
          text="Log out"
          icon="exit_to_app"
          href="/logout"
          color="#D33649"
        />
      </LinksContainer>
    </DropdownMenuList>
  </DropdownMenuContainer>
);

export default DropdownMenu;
