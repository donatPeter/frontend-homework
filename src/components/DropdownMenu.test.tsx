import React from "react";
import { shallow } from "enzyme";
import DropdownMenu from "./DropdownMenu";

describe("<DropdownMenu />", () => {
  it("renders menu", () => {
    expect(shallow(<DropdownMenu />)).toMatchSnapshot();
  });
});
