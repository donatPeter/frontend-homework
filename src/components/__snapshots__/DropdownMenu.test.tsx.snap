// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownMenu /> renders menu 1`] = `
<styled.div>
  <styled.div>
    <Connect(Companies) />
    <styled.div>
      <MenuLink
        href="/get-app"
        icon="phone_iphone"
        text="Get the mobile app"
      />
      <MenuLink
        href="/community"
        icon="people"
        text="Community"
      />
      <MenuLink
        href="/help"
        icon="book"
        text="Knowledge base"
      />
    </styled.div>
    <styled.div>
      <MenuLink
        href="/settings"
        icon="settings"
        text="Settings"
      />
      <MenuLink
        color="#D33649"
        href="/logout"
        icon="exit_to_app"
        text="Log out"
      />
    </styled.div>
  </styled.div>
</styled.div>
`;
